src/Services/UserService.cs

```diff
@@ -1,20 +1,51 @@
 using Coffee_Shop_App.src.Abstractions;
 using Coffee_Shop_App.src.Entities;
-using Coffee_Shop_App.src.Repositories;
 using Coffee_Shop_Appe.src.Abstractions;
 
-namespace Coffee_Shop_App.src.Services;
+namespace Coffee_Shop_App;
 
+public class UserService : IUserService
+{
 
+    private IUserRepository? _userRepository; //to talk to the Repo
 
-public class UserService 
-{
+    public UserService(IUserRepository? userRepository) //constructor DI 
+    {
+        _userRepository = userRepository;
+    }
+
+
+
+
+    public List<User> FindAll()
+    {
+        return _userRepository.FindAll(); //to talk to the Repo
+    }
 
-    public string SayHi(){
-        var helloRepository = new UserRepository();
-        return helloRepository.SayHi();
 
+
+    public User CreateOne(User user)
+    {
+        // User? foundUser = _userRepository.findOne(userId);
+
+        // if (foundUser is not null)
+        // {
+        //     return null;
+        // }
+
+        return _userRepository.CreateOne(user);
+    }
+
+
+    public User? findOne(string userId)
+    {
+
+        return _userRepository.findOne(userId);
     }
 
+    public User? findOneByEmail(string userEmail)
+    {
 
-}
+        return _userRepository.findOne(userEmail);
```

Comment:
here you have to call `findOneByEmail()` method from the `_userRepository` instead of `findOne()`